### PR TITLE
Do not require target dir in mb makehashes

### PR DESCRIPTION
**Pull request by andrii-suse**

### CODE REVIEW

File: mb/mb/hashes.py [hunk 1]
```diff
@@ -51,7 +51,9 @@ def __init__(self, basename, src_dir=None, dst_dir=None,
         self.dst_dir = dst_dir
 
         self.dst_basename = '%s.size_%s' % (self.basename, self.size)
-        self.dst = os.path.join(self.dst_dir, self.dst_basename)
+        self.dst = None
+        if dst_dir:
```
Comment (darix) [inline mb/mb/hashes.py]: shouldnt this be self.dst_dir ?

File: mb/scripts/mb [hunk 2]
```diff
@@ -1000,7 +1000,7 @@ class MirrorDoctor(cmdln.Cmdln):
                         help='set the base directory (so that you can work on a '
                              'subdirectory -- see examples)')
     @cmdln.option('-t', '--target-dir', metavar='PATH',
-                        help='set the target directory (required)')
+                        help='set the target directory (optional, so file hashes can be created as well)')
```
Comment (darix) [inline mb/scripts/mb]: This is deprecated. it was announced to be deprecated in 201X already!

File: mb/mb/hashes.py (hunk 1 above)
Comment (andrii-suse) [reply mb/mb/hashes.py]: they are equal (see line 51), I don't see difference

File: mb/scripts/mb (hunk 2 above)
Comment (andrii-suse) [reply mb/scripts/mb]: It still doesn't hurt to have it optional. So e.g. hashes are available outside of DB as well.

Comment (darix) [reply mb/scripts/mb]: really ... stuff that has 5+ years of announcement that is deprecated can go.

**Pull request merged**